docs/examples/example_onnx.ipynb

```diff
@@ -0,0 +1,337 @@
+    "\n",
+    "df = pd.read_csv(git_root(\"data/learning_mindset.zip\"))\n",
+    "outcome_column = \"achievement_score\"\n",
+    "treatment_column = \"intervention\"\n",
+    "feature_columns = [\n",
+    "    column for column in df.columns if column not in [outcome_column, treatment_column]\n",
+    "]\n",
+    "categorical_feature_columns = [\n",
+    "    \"ethnicity\",\n",
+    "    \"gender\",\n",
+    "    \"frst_in_family\",\n",
+    "    \"school_urbanicity\",\n",
+    "    \"schoolid\",\n",
+    "]\n",
+    "# Note that explicitly setting the dtype of these features to category\n",
+    "# allows both lightgbm as well as shap plots to\n",
+    "# 1. Operate on features which are not of type int, bool or float\n",
+    "# 2. Correctly interpret categoricals with int values to be\n",
+    "#    interpreted as categoricals, as compared to ordinals/numericals.\n",
+    "for categorical_feature_column in categorical_feature_columns:\n",
+    "    df[categorical_feature_column] = df[categorical_feature_column].astype(\"category\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we've loaded the experiment data, we can train a MetaLearner.\n",
+    "\n",
+    "\n",
+    "### Training a MetaLearner\n",
+    "\n",
+    "Again, mirroring our {ref}`example on estimating CATEs with a MetaLearner\n",
+    "<example-basic>`, we can train an\n",
+    "{class}`~metalearners.XLearner` as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from metalearners import XLearner\n",
+    "from lightgbm import LGBMRegressor, LGBMClassifier\n",
+    "\n",
+    "xlearner = XLearner(\n",
+    "    nuisance_model_factory=LGBMRegressor,\n",
+    "    propensity_model_factory=LGBMClassifier,\n",
+    "    treatment_model_factory=LGBMRegressor,\n",
+    "    is_classification=False,\n",
+    "    n_variants=2,\n",
+    "    nuisance_model_params={\"n_estimators\": 5, \"verbose\": -1},\n",
+    "    propensity_model_params={\"n_estimators\": 5, \"verbose\": -1},\n",
+    "    treatment_model_params={\"n_estimators\": 5, \"verbose\": -1},\n",
+    "    n_folds=2,\n",
+    ")\n",
+    "\n",
+    "xlearner.fit(\n",
+    "    X=df[feature_columns],\n",
+    "    y=df[outcome_column],\n",
+    "    w=df[treatment_column],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Converting the base models to ONNX\n",
+    "\n",
+    "Before being able to convert the MetaLearner to ONXX we need to manually convert the necessary\n",
+    "base models for the prediction. To get a list of the necessary base models that need to be\n",
+    "converted we can use :meth:`~metalearners.MetaLearner.necessary_onnx_models`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xlearner.necessary_onnx_models()"
```

Comment:
Would it make sense to keep this member function private for now? Furhtermore, might it make sense to have this function return a `dict[str, list[Model]]` where the key of the dict is the model name and `Model` is the thing with the `_overall_estimator` member. This would avoid the need of having three for loops with private functions below.

Reply:
Good idea! Done!